src/emitter.ts

```diff
@@ -0,0 +1,1126 @@
+        if (prefix === "") {
+            return `this: ${i.name}, `;
+        }
+        else {
+            return pollutor ? `this: ${pollutor.name}, ` : "";
+        }
+    }
+
+    // A covariant  EventHandler is one that is defined in a parent interface as then redefined in current interface with a more specific argument types
+    // These patterns are unsafe, and flagged as error under --strictFunctionTypes.
+    // Here we know the property is already defined on the interface, we elide its declaration if the parent has the same handler defined
+    function isCovariantEventHandler(i: Browser.Interface, p: Browser.Property) {
+        return isEventHandler(p) &&
+            iNameToEhParents[i.name] && iNameToEhParents[i.name].length > 0 &&
+            !!iNameToEhParents[i.name].find(
+                i => iNameToEhList[i.name] && iNameToEhList[i.name].length > 0 &&
+                    !!iNameToEhList[i.name].find(e => e.name === p.name));
+    }
+
+    function emitProperty(prefix: string, i: Browser.Interface, emitScope: EmitScope, p: Browser.Property, conflictedMembers: Set<string>) {
+        function printLine(content: string) {
+            if (conflictedMembers.has(p.name)) {
+                printer.printLineToStack(content);
+            }
+            else {
+                printer.printLine(content);
+            }
+        }
+
+        emitComments(p, printLine);
+
+        // Treat window.name specially because of https://github.com/Microsoft/TypeScript/issues/9850
+        if (p.name === "name" && i.name === "Window" && emitScope === EmitScope.All) {
+            printLine("declare const name: never;");
+        }
+        else {
+
+            let pType: string;
+            if (p["override-type"]) {
+                pType = p["override-type"]!;
+            }
+            else {
+                if (isEventHandler(p)) {
+                    // Sometimes event handlers with the same name may actually handle different
+                    // events in different interfaces. For example, "onerror" handles "ErrorEvent"
+                    // normally, but in "SVGSVGElement" it handles "SVGError" event instead.
+                    const eType = p["event-handler"] ? getEventTypeInInterface(p["event-handler"]!, i) : "Event";
+                    pType = `(${emitEventHandlerThis(prefix, i)}ev: ${eType}) => any`;
+                }
+                else {
+                    pType = convertDomTypeToTsType(p);
+                }
+            }
+            const requiredModifier = !p.required || p.required === "1" ? "" : "?";
+            pType = p.nullable ? makeNullable(pType) : pType;
+            const readOnlyModifier = p["read-only"] && prefix === "" ? "readonly " : "";
+            printLine(`${prefix}${readOnlyModifier}${p.name}${requiredModifier}: ${pType};`);
+        }
+    }
+
+    function emitComments(entity: { comment?: string; deprecated?: 1 } | undefined, print: (s: string) => void) {
```

Comment:
Don't think it's helpful to handle undefined input here -- in two of three callsites the input is definitely defined, in the other we're  calling `emitComments` immediately before an `if` and could just move it into the `if`.